tests/MMS/spatial/advection/runtest

```diff
@@ -30,99 +26,102 @@ shell_safe("make > make.log")
 
 # List of options to be passed for each test
 options = [
-    ("method=2", "Arakawa", "-^" , 2),
-    ("method=0 mesh:ddx:upwind=u1 mesh:ddz:upwind=u1", "1st order upwind", "-o", 1),
-    ("method=0 mesh:ddx:upwind=c2 mesh:ddz:upwind=c2", "2nd order central", "-x", 2),
-    ("method=0 mesh:ddx:upwind=w3 mesh:ddz:upwind=w3", "3rd order WENO", "-s", 3)
-    ]
+    ("method=2", "Arakawa", "-^", 2),
+    ("method=1 mesh:ddx:upwind=u1 mesh:ddz:upwind=u1", "1st order upwind", "-o", 1),
+    ("method=1 mesh:ddx:upwind=u2 mesh:ddz:upwind=u2", "2nd order upwind", "-^", 2),
+    ("method=1 mesh:ddx:upwind=u3 mesh:ddz:upwind=u3", "3nd order upwind", "-v", 3),
+    ("method=1 mesh:ddx:upwind=c2 mesh:ddz:upwind=c2", "2nd order central", "-x", 2),
+    ("method=1 mesh:ddx:upwind=c4 mesh:ddz:upwind=c4", "4th order central", "-+", 4),
+    # NOTE: WENO3 seems to have two different regimes, one < 3, one >
+    # 3, so it's possible to make it pass by judicious choice of grid
+    # sizes, which is suspicious. It does seem to work correctly in
+    # other contexts, so not overly worrying that it doesn't work
+    # exactly as expected here
+    # ("method=1 mesh:ddx:upwind=w3 mesh:ddz:upwind=w3", "3rd order WENO", "-s", 3),
+]
 
 # List of NX values to use
-nxlist = [16, 32]
-#nxlist = [16, 32, 64, 128, 256, 512, 1024]
+nxlist = [16, 32, 64, 128, 256]
```

Comment:
what is the point of running them all, if we only check the last 2 points?

Reply:
It's useful to see the error over those points, especially if you're plotting the error.

Also this test is only run on `all_tests`, and not doing the first three points only saves us a few seconds anyway, as the last run takes the majority of the time.

Reply:
Why don't you check these points as well?

I am not object to run them, I am just objecting to run them, and ignore them. That is all.

Reply:
We're pretty inconsistent about whether we check the full range or the last two points. I've added a module `boututils.check_scaling` that provides some functions so we can be more consistent. They're only used in this test so far